IdentityCore/src/webview/systemWebview/MSIDSafariViewController.m

```diff
@@ -0,0 +1,130 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSIDSafariViewController.h"
+#import "MSIDSystemWebviewController.h"
+#import <SafariServices/SafariServices.h>
+#import "MSIDWebOAuth2Response.h"
+
+@interface MSIDSafariViewController() <SFSafariViewControllerDelegate>
+
+@end
+
+@implementation MSIDSafariViewController
+{
+    SFSafariViewController *_safariViewController;
+    
+    NSURL *_url;
+
+    id<MSIDRequestContext> _context;
+}
+
+- (id)initWithURL:(NSURL *)url
+          context:(id<MSIDRequestContext>)context
+{
+    self = [super init];
+    if (self)
+    {
+        _url = url;
+        _context = context;
+        
+        _safariViewController = [[SFSafariViewController alloc] initWithURL:url entersReaderIfAvailable:NO];
```

Comment:
how do we ensure that we're on the main thread?

Reply:
Good point, I think it is logical that we simply check if we are on main thread when start is called, and return NO if we aren't.